src/entity.ts

```diff
@@ -63,17 +64,21 @@ export class Entity {
 			}
 
 			// Raise the initNew or initExisting event on this type and all base types
-			context.whenReady(() => {
-				for (let t = type; t; t = t.baseType) {
-					if (isNew)
-						(t.initNew as Event<Type, EntityInitNewEventArgs>).publish(t, { entity: this });
-					else
-						(t.initExisting as Event<Type, EntityInitExistingEventArgs>).publish(t, { entity: this });
-				}
+			this.initialized = new Promise(resolve => {
+				context.whenReady(() => {
+					for (let t = type; t; t = t.baseType) {
+						if (isNew)
+							(t.initNew as Event<Type, EntityInitNewEventArgs>).publish(t, { entity: this });
+						else
+							(t.initExisting as Event<Type, EntityInitExistingEventArgs>).publish(t, { entity: this });
+					}
 
-				// Set values of new entity for provided properties
-				if (isNew && properties)
-					this.updateWithContext(context, properties);
+					// Set values of new entity for provided properties
+					if (isNew && properties)
+						this.updateWithContext(context, properties);
+
+					context.whenReady(resolve);
+				});
```

Comment:
Because entity initialization and updates always have the possibility for async value resolution, the `initialized` promise provides a way to wait for the instance to be "fully baked" if that's something the user needs. As evidenced by most of the entity unit tests, if there is no asynchrony involved with the construction of a particular entity, a caller can assume the entity is baked immediately after constructing it.

Reply:
I had the thought that it might be nice to not expose it if a user would likely never have need to use the `initialized` promise, and handle it (somehow) via the `create` method. But, I don't think having the `initialized` property is a bad thing per se, and it probably isn't worth the effort (assuming it would be feasible).